### ISSUE TITLE
`@stylistic/function-parentheses-newline-inside` reads a value against the comment spans it had before its own fix, so `always` takes two runs

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#288]
Found while sweeping the value shapes of that rule for #281, and reproduced on `main` unchanged, so the branch neither adds it nor touches it.

The rule finds the spans the inline comments of a value occupy once, before it walks the value, and every function of that value is then read against those spans. A fix written for one function moves the text of every function behind it, and the spans are not moved with it: where the first fix closes a comment earlier than the file did, the second function is read as standing inside a comment it no longer stands inside, and the problem it has is reported only on the next run.

### What minimal example or steps are needed to reproduce the bug?

```less
a { t: foo(1px // c) calc(/*b*/
 ,2px
); }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
	"customSyntax": "postcss-less",
	"plugins": "@stylistic/stylelint-plugin",
	"rules": {
		"@stylistic/function-parentheses-newline-inside": "always"
	}
}
```

### What did you expect to happen?

One run of `--fix` leaving a file the rule has nothing more to say about, which is what the rule does for every other shape of this value.

### What actually happened?

The first run breaks `foo(` on the inside of both its parentheses and says nothing about `calc(`, and the second run breaks `calc(` as well. Under `postcss-scss` the same. The third run settles, so nothing runs away — the file a user gets from one run is simply one break short of what the option asks for.

### Do you have a proposal to fix the bug?

The spans come from `findInlineCommentSpans(declValue, …)`, read once per declaration, and the walk over the parsed value writes into the nodes as it goes. On the first run the double slash of `foo(1px // c)` opens a comment that runs to the first line break of the value, which stands inside `calc(`, so the whitespace `calc(` keeps in front of its first argument is read as the text of that comment and the `always` option is satisfied by it. Breaking `foo(` on the inside writes a line break in front of `c`, which closes that comment where the file did not, and `calc(` is left with whitespace holding no break at all.

Either the spans are found again wherever a fix has written into the value, or the value is walked once per fix; the first is the cheaper of the two, since the fixes of this rule are written into the parsed tree and the value is printed once at the end.

**Comment by firefoxic** [comment]
The last sentence of the proposal is out of date since #272 arrived as `41c40a6`. This rule no longer writes its fixes into the parsed tree, and no longer prints the value at the end: each fix names the span of the value it changes, and the spans are written through `applyEditsFromEnd` once the walk is over.

Nothing about the bug itself moves. The spans of the inline comments are still found once, before the walk, and a fix written for one function still moves the text of every function behind it, so the second function is still read against a comment span the value no longer has. `a { t: foo(1px // c) calc(/*b*/⏎ ,2px⏎); }` under `always` still takes two runs, under `postcss-less` and `postcss-scss` alike.

What changes is the cost of the two ways out this issue names, and the first is now cheaper than it was rather than dearer. Every fix is a `{ start, end, text }` counted in the value the file spells, so what a fix moves, and by how much, is a subtraction rather than something to be recovered from the tree — finding the spans again after a write is arithmetic on the edits already collected, and no reparse is wanted for it.
